src/Http/Routing/src/RouteEndpointDataSource.cs

```diff
@@ -232,6 +236,15 @@ private RouteEndpointBuilder CreateRouteEndpointBuilder(
             entrySpecificConvention(builder);
         }
 
+        // Initialize this route endpoint builder with validation convention if validation options
+        // are registered and validation is not disabled on the endpoint.
+        var hasValidationOptions = builder.ApplicationServices.GetService(typeof(IOptions<ValidationOptions>)) is not null;
```

Comment:
With the default MEDI, this will always be non-null.

Reply:
I've updated this and the below to no-op if there are no resolvers registered. The better thing to do might be to add a marker interface that we can assert on but meh on marker interfaces.